### PR TITLE
Handling one or no images in click handler

### PR DESCRIPTION
**Pull request by mlunoe**

### CODE REVIEW

File: src/js/components/modals/ImageViewerModal.js [hunk 1]
```diff
@@ -39,8 +39,11 @@ class ImageViewerModal extends React.Component {
   }
 
   handleKeyPress(event) {
-    if (event.defaultPrevented) {
-      return; // Should do nothing if the key event was already consumed.
+    let {images} = this.props;
+    // Should do nothing if the key event was already consumed, or there is only
+    // one or no images
+    if (event.defaultPrevented || images || images.length < 1) {
```
Comment (kennyt) [inline src/js/components/modals/ImageViewerModal.js]: if the `images` variable is an array, won't this always return true?

Comment (mlunoe) [inline src/js/components/modals/ImageViewerModal.js]: Oh man! Thanks!

File: src/js/components/modals/ImageViewerModal.js [hunk 2]
```diff
@@ -38,9 +38,24 @@ class ImageViewerModal extends React.Component {
...
 
+  componentWillUpdate(nextProps, {isLoadingImage}) {
+    let {images, selectedImage} = this.props;
+    let nextImages = nextProps.images;
+    let nextSelectedImage = nextProps.selectedImage;
+    // Two images in a row are the same, img.onLoad will never be called, so
+    // here we are updating manually
+    if (isLoadingImage &&
+        images[selectedImage] === nextImages[nextSelectedImage]) {
+      this.handleLoadingImageChange(false);
```
Comment (kennyt) [inline src/js/components/modals/ImageViewerModal.js]: the react docs stress that we shouldn't use `setState` inside of `componentWillUpdate` and to instead do it in `componentWillReceiveProps`. is this possible?

Comment (mlunoe) [inline src/js/components/modals/ImageViewerModal.js]: It is not, it comes at the wrong timing. Before we have set state to loading, unfortunately 😢

File: src/js/components/modals/ImageViewerModal.js [hunk 3]
```diff
@@ -38,9 +38,24 @@ class ImageViewerModal extends React.Component {
...
   }
 
+  componentWillUpdate(nextProps, {isLoadingImage}) {
+    let {images, selectedImage} = this.props;
+    let nextImages = nextProps.images;
+    let nextSelectedImage = nextProps.selectedImage;
+    // Two images in a row are the same, img.onLoad will never be called, so
+    // here we are updating manually
+    if (isLoadingImage &&
+        images[selectedImage] === nextImages[nextSelectedImage]) {
```
Comment (kennyt) [inline src/js/components/modals/ImageViewerModal.js]: this is indented differently than we usually do

File: src/js/components/modals/ImageViewerModal.js (hunk 2 above)
Comment (kennyt) [inline src/js/components/modals/ImageViewerModal.js]: I just thought of something. I'm pretty sure if you add a `key` property on the `img` tag and make it different for every single image (including the same image), the `onLoad` will trigger again and we can evade all of this

Comment (mlunoe) [inline src/js/components/modals/ImageViewerModal.js]: Nice! That was a really good idea! Thanks dude!

**Pull request merged**